### PR TITLE
1110: Rebase and Reorganize IBM Oem schemas

### PR DESCRIPTION
**Pull request by baemyung**
In summary, This PR renames IBM schemas like

- OemAssembly to openbmc/OpenBMCAssembly
- OemLogEntry to openbmc/OpenBMCLogEntry
- OemMessage to openbmc/OpenBMCMessage
- OemComputerSystem to ibm/IBMComputerSystem (for IBM part)
- OemFabricAdapter to ibm/IBMFabricAdapter
- OemManagerAccount to ibm/IBMManagerAccount
- OemLogEntryAttachement to ibm/IBMLogEntryAttachement
- OemPCIeDevice to ibm/IBMPCIeDevice
- OemPCIeSlots to ibm/IBMPCIeSlots
- OemServiceRoot to ibm/IBMServiceRoot

As part of this, the following will be changed.
- `/redfish/v1/Systems/system/Actions/Oem/OemComputerSystem.ExecutePanelFunction` to `/redfish/v1/Systems/system/Actions/Oem/IBM/IBMComputerSystem.ExecutePanelFunction`.

WebUI will also need to change  
- https://github.com/ibm-openbmc/webui-vue/blob/f1a5c793f27c2198ae13df341715134e13e0b6f5/src/store/modules/Logs/IBMiServiceFunctionsStore.js#L26

However, the following IBM schemas are kept under the same Oem schemas

**- OemUpdateService**
OemUpdateService is being via `/redfish/v1/UpdateService/Actions/Oem/OemUpdateService.ConcurrentUpdate` - https://github.com/ibm-openbmc/bmcweb/blob/f27f6e8ab4bd8453f925cc882307efaa5c87b0be/redfish-core/lib/update_service.hpp#L894

As this routing point is used by HMC, it will not be touched at this time.

-----------

In detail, the following upstream commits are brought in.

- https://github.com/openbmc/bmcweb/commit/1d19d872 //Fix Missed Renamed OpenBMCManager word
- https://github.com/openbmc/bmcweb/commit/e3142e80 //Remove unused OemComputerSystem json file
- https://github.com/openbmc/bmcweb/commit/7805ca2f //Fix OpenBMC JsonSchemas installation
- https://github.com/openbmc/bmcweb/commit/c44cddd8 //Format OEM schemas with xmllint
- https://github.com/openbmc/bmcweb/commit/f3025d9f //Correct OpenBMCComputerSystem schema content
- https://github.com/openbmc/bmcweb/commit/2401e814 //Remove old symlinks
- https://github.com/openbmc/bmcweb/commit/d1eeb791 //Update copyright and schema location
- https://github.com/openbmc/bmcweb/commit/1d834d49 //OemComputerSystem -> OpenBMCComputerSystem
- https://github.com/openbmc/bmcweb/commit/f958ed9c //Rename OemVirtualMedia to OpenBMCVirtualMedia
- https://github.com/openbmc/bmcweb/commit/fc1cdd14 //Fix OpenBMC OEMManager

Additionally, the following 1110 downstream-only commits are added.

- Rename IBM OemSchemas to IBMSchemas
- Separate IBM OemComputerSystem from OpenBMC schema
- Separate IBMManager from OpenBMCManager
- Relocate IBM schemas

---

### CODE REVIEW

File: redfish-core/lib/log_services.hpp [hunk 1]
```diff
@@ -2838,7 +2838,7 @@ inline void requestRoutesDBusEventLogEntry(App& app)
             if constexpr (BMCWEB_IBM_MANAGEMENT_CONSOLE)
             {
                 asyncResp->res.jsonValue["Oem"]["OpenBMC"]["@odata.type"] =
```
Comment (gtmills) [inline redfish-core/lib/log_services.hpp:2840]: The problem here is this is still under OpenBMC

File: redfish-core/lib/log_services.hpp [hunk 2]
```diff
@@ -3071,7 +3071,7 @@ inline void requestRoutesDBusCELogEntry(App& app)
                     *notifyAction;
             }
             asyncResp->res.jsonValue["Oem"]["IBM"]["@odata.type"] =
-                "#OemLogEntryAttachment.v1_0_0.IBM";
```
Comment (gtmills) [inline redfish-core/lib/log_services.hpp:3074]: This works well because it is under IBM

File: redfish-core/lib/assembly.hpp [hunk 3]
```diff
@@ -286,7 +286,7 @@ void getAssemblyPresence(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
             if (fru == "panel0" || fru == "panel1")
             {
                 data["Oem"]["OpenBMC"]["@odata.type"] =
-                    "#OemAssembly.v1_0_0.Assembly";
```
Comment (gtmills) [inline redfish-core/lib/assembly.hpp:289]: This is under the OpenBMC name space but an IBM schema?

Comment (baemyung) [reply redfish-core/lib/assembly.hpp:289]: > This is under the OpenBMC name space but an IBM schema?

Ok, we can modify IBMAssembly schema - to rename `oem/OpenBMC/ReadyToRemove` to `oem/IBM/ReadyToRemove`.

File: redfish-core/lib/log_services.hpp (hunk 1 above)
Comment (baemyung) [reply redfish-core/lib/log_services.hpp:2840]: I'll rename `Oem/OpenBMC/ `to` Oem/IBM/ `for IBMLogEntry schema

Comment (baemyung) [reply redfish-core/lib/log_services.hpp:2840]: I'll keep `Oem/OpenBMC/`  and move IBMLogEntry.json to `oem/openbmc/json-schema/OpenBMCLogEntry.json`

File: redfish-core/lib/log_services.hpp (hunk 2 above)
Comment (baemyung) [reply redfish-core/lib/log_services.hpp:3074]: Yes.

File: redfish-core/lib/assembly.hpp (hunk 3 above)
Comment (baemyung) [reply redfish-core/lib/assembly.hpp:289]: I'll keep them as they were originally `Oem/OpenBMC/ReadyToRemove `and put the schema file as `OpenBMCAssembly.json`

File: redfish-core/schema/oem/openbmc/json-schema/OpenBMCManager.v1_0_0.json [hunk 4]
```diff
@@ -256,55 +298,16 @@
             "properties": {},
             "type": "object"
         },
-        "Oem": {
```
Comment (gtmills) [inline redfish-core/schema/oem/openbmc/json-schema/OpenBMCManager.v1_0_0.json:259]: Shouldn't these go in an IBMManager?

Comment (baemyung) [reply redfish-core/schema/oem/openbmc/json-schema/OpenBMCManager.v1_0_0.json:259]: OpenBMCManager.json is the same as upstream OpenBMCManager.json - https://github.com/openbmc/bmcweb/blob/6be832e2963e9d720dd95543358eca380c5e52d2/redfish-core/schema/oem/openbmc/json-schema/OpenBMCManager.v1_0_0.json#L2

IBM part is moved out of OpenBMC schem to IBM schema namespace.

File: redfish-core/schema/oem/ibm/csdl/OemComputerSystem_v1.xml [hunk 5]
```diff
@@ -0,0 +1,101 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (gtmills) [inline redfish-core/schema/oem/ibm/csdl/OemComputerSystem_v1.xml:1]: since are all IBM, can we call this and the json IBMComputerSystem

**Review by gtmills — APPROVED** [review]

**Comment by baemyung** [comment]
The tracking Jira story is https://jsw.ibm.com/browse/PFEBMC-3330

**Pull request merged**